### PR TITLE
Rewrote Terminal from the beginning

### PR DESCRIPTION
**Pull request by SmooZzzie**
I rewrote Terminal from the beginning due to my awful project acrhitecture and NOT OOP code (couldn't change something without destroying some other blocks of my program).

Added filter and a simple logger. Waiting for review.

### CODE REVIEW

File: ArtemShchepets/src/main/java/week1/controller/ITerminalControllerFactory.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package week1.controller;
+
+import week1.database.IAppDBImpl;
+import week1.interfaces.IAppDB;
+import week1.interfaces.ITerminalController;
+import week1.model.Seller;
+
+public class ITerminalControllerFactory {
+
+    public static ITerminalController create() {
```
Comment (sasha943) [inline ArtemShchepets/src/main/java/week1/controller/ITerminalControllerFactory.java]: create(IAppDB iAppDB); create method should only return instance of ITerminalControllerImpl. Sellers should not be present

File: ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+    public ITerminalControllerImpl(IAppDB iAppDB) {
+        this.iAppDB = iAppDB;
+        logger.setLevel(Level.OFF);
+    }
+
+    @Override
+    public boolean login(String login, String password) {
+
+        iAppDB.setCurrentSeller(-1);
```
Comment (sasha943) [inline ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java]: -1? What is means

File: ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+        iAppDB.updateBill(bill);
+
+        logger.info("Bill was closed!");
+        return bill;
+    }
+
+    @Override
+    public Bill findBillById(int billId) {
+
```
Comment (sasha943) [inline ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java:123]: add validation

Comment (SmooZzzie) [reply ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java:123]: What kind of validation? if id < 0 or what?

File: ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java (hunk 2 above)
Comment (SmooZzzie) [reply ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java]: currentSeller = -1 is a state to show that no one is logged in now. Will refactor this code to more understandable.

File: ArtemShchepets/src/main/java/week1/comparators/SellersSoldProductsComparator.java [hunk 4]
```diff
@@ -11,4 +11,5 @@
     public int compare(Seller o1, Seller o2) {
```
Comment (sasha943) [inline ArtemShchepets/src/main/java/week1/comparators/SellersSoldProductsComparator.java]: not valid compare) if they sold the same quantity of product?  Better by fullName

File: ArtemShchepets/src/main/java/week1/controller/ITerminalControllerFactory.java (hunk 1 above)
Comment (SmooZzzie) [reply ArtemShchepets/src/main/java/week1/controller/ITerminalControllerFactory.java]: return new ITerminalControllerImpl( new IAppDBImpl()); - like that?

File: ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java (hunk 3 above)
Comment (SmooZzzie) [reply ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java:123]: https://github.com/presly808/aco22/pull/51/commits/b45056f67561831e9fe77997114005ebcd0e7a40 Fixed

File: ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java (hunk 2 above)
Comment (SmooZzzie) [reply ArtemShchepets/src/main/java/week1/controller/ITerminalControllerImpl.java]: https://github.com/presly808/aco22/pull/51/commits/b45056f67561831e9fe77997114005ebcd0e7a40 Fixed

File: ArtemShchepets/src/main/java/week1/controller/ITerminalControllerFactory.java (hunk 1 above)
Comment (SmooZzzie) [reply ArtemShchepets/src/main/java/week1/controller/ITerminalControllerFactory.java]: Fixedhttps://github.com/presly808/aco22/pull/51/commits/b45056f67561831e9fe77997114005ebcd0e7a40 Fixed

Comment (SmooZzzie) [reply ArtemShchepets/src/main/java/week1/controller/ITerminalControllerFactory.java]: https://github.com/presly808/aco22/pull/51/commits/b45056f67561831e9fe77997114005ebcd0e7a40 Fixed

**Comment by SmooZzzie** [comment]
Can you tell me if I use all design patterns right or not, pls?

File: ArtemShchepets/src/main/java/week1/controller/ProxyITerminalControllerImpl.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    @Override
+    public List<Bill> filter(LocalDateTime startTime, LocalDateTime endTime, Comparator<Bill> comparator) {
+
+        System.out.println("[" + LocalTime.now() + "]: User is trying to filter some bills.");
+
+        return terminalController.filter(startTime, endTime, comparator);
+    }
+
+    @Override
+    public void turnOnTerminalLogger() {
```
Comment (sasha943) [inline ArtemShchepets/src/main/java/week1/controller/ProxyITerminalControllerImpl.java:106]: move to utils

File: ArtemShchepets/src/main/java/week1/interfaces/IAppDB.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+import week1.model.Bill;
+import week1.model.Seller;
+
+import java.util.List;
+import java.util.logging.Logger;
+
+public interface IAppDB {
+
+    Logger getLogger();
```
Comment (sasha943) [inline ArtemShchepets/src/main/java/week1/interfaces/IAppDB.java:11]: Do not need this. Create class MyLogger with method: public void writeLog(String message)

**Pull request merged**